.circleci/config.yml

```diff
@@ -188,79 +194,92 @@ jobs:
           command: |
             python -m venv env
             . env/bin/activate
-            pip install dist/dwave-gate*.tar.gz
+            pip install dist/dwave_gate*.tar.gz
       - run:
           name: run tests
           command: |
             . env/bin/activate
             pip install -r requirements_dev.txt
             pytest tests/
 
+  test-dependencies:
+    parameters:
+      python-version:
+        type: string
+      dependency-versions:
+        type: string
+
+    docker:
+      - image: python:<< parameters.python-version >>-slim
+
+    steps:
+      - checkout
+      - attach_workspace:
+          at: dist
+      - run:
+          name: install
+          command: |
+            python -m venv env
+            . env/bin/activate
+            pip install << parameters.dependency-versions >> --upgrade --only-binary=numpy
+            pip install dwave-gate --no-index -f dist/ --no-deps --force-reinstall
+      - run: &unix-run-tests
+          name: run tests
+          command: |
+            . env/bin/activate
+            pip install -r requirements_dev.txt
+            pytest tests/
 workflows:
   tests:
     jobs:
-      - build-and-test-linux: &build
-          matrix:
-            parameters:
-              python-version: &python-versions [3.8.9, 3.9.4, 3.10.0, 3.11.0, 3.12.0]
-      - build-and-test-linux-aarch64:
+      - build-linux: &build
           matrix:
             parameters:
-              python-version: *python-versions
+              python-version: &python-versions [3.8.19, 3.9.19, 3.10.14, 3.11.9, 3.12.4]
+      - build-linux-aarch64: *build
       - build-sdist
-      - build-and-test-osx: &build-and-test-osx
+      - build-osx: *build
+      - build-windows: *build
+      - test-codecov
+      - test-dependencies:
+          name: test-dependencies - << matrix.dependency-versions >> - py << matrix.python-version >>
+          requires:
+            - build-linux
           matrix:
             parameters:
+              # test the lowest and highest for each dependency
+              dependency-versions: [numpy==1.21.6, numpy]
```

Comment:
@arcondello How important is it that we support `oldest-supported-numpy` vs. just setting a lower limit manually (in this case to support the typing module). I'd assume this is fine with regards to the SDK?

Reply:
Yes, that's fine. As long as you're testing the full range.